mavros_extras/src/plugins/odom.cpp

```diff
@@ -50,6 +49,43 @@ class OdometryPlugin : public plugin::PluginBase {
 
 		// subscribers
 		odom_sub = odom_nh.subscribe("odom", 10, &OdometryPlugin::odom_cb, this);
+
+		//! Map from param string to local MAV_FRAME
+		const std::unordered_map<std::string, MAV_FRAME> lf_map{
+			{ "vision_enu", MAV_FRAME::VISION_ENU },
+			{ "vision_ned", MAV_FRAME::VISION_NED },
+			{ "mocap_enu", MAV_FRAME::MOCAP_ENU },
+			{ "mocap_ned", MAV_FRAME::MOCAP_NED }
+		};
+
+		//! Map from param string to body MAV_FRAME
+		const std::unordered_map<std::string, MAV_FRAME> bf_map{
+			{ "ned", MAV_FRAME::BODY_NED },
+			{ "frd", MAV_FRAME::BODY_FRD },
+			{ "flu", MAV_FRAME::BODY_FLU }
+		};
+
+		auto local_frame_it = lf_map.find(local_frame);
+		auto body_frame_it = bf_map.find(body_frame_orientation);
+
+		/** Determine frame_id naming - considering the ROS msg frame_id
+		 * as "odom" by default
+		 */
+		if (local_frame_it == lf_map.end()) {
+			ROS_ERROR_NAMED("odom", "Odometry: invalid local frame \"%s\"", local_frame.c_str());
+			return;
```

Comment:
Not return. Set to some default.

Reply:
Actually, I don't want it to send the wrong frame_id to the flight controller. That's why this is an error and not a warning.